### ISSUE TITLE
Ep 23: Drop python 3.8 from build matrix

### ISSUE THREAD

**Issue by gcapes** [carpentries-incubator/python-intermediate-development#253]
My requirements.txt file contains matplotlib=3.8.0, which doesn't seem to be found for python 3.8.

My failed GA run here: https://github.com/gcapes/python-intermediate-inflammation/actions/runs/6718827862
and the subsequent run with 3.8 removed here: https://github.com/gcapes/python-intermediate-inflammation/actions/runs/6718965580

The build matrix then looks like this:
```
strategy:
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
        python-version: ["3.9", "3.10"]

    runs-on: ${{ matrix.os }}
```

**Comment by anenadic** [comment]
This one may be a good one for you @douglowe as it is relevant to the [NERC version of the course ](https://github.com/carpentries-incubator/python-intermediate-development-earth-sciences)as well?

**Comment by anenadic** [comment]
Related to #274 and fixed via #274.
